c_emulator/rvfi_dii.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "sail.h"
+
+typedef enum {
+  RVFI_prestep_continue,  // continue loop
+  RVFI_prestep_eof,       // Got EOF, delete rvfi and return
+  RVFI_prestep_end_trace, // Got EndOfTrace
+  RVFI_prestep_ok,        // Ready for step
+} rvfi_prestep_t;
+
+typedef void (*packet_reader_fn)(lbits *rop, unit);
+
+class rvfi_t {
```

Comment:
```suggestion
class rvfi_handler {
```
Maybe something like this? But I don't really mind what name we choose.